src/distribution.cpp

```diff
@@ -111,6 +107,32 @@ void Discrete::normalize()
   }
 }
 
+//==============================================================================
+// Discrete implementation
+//==============================================================================
+
+Discrete::Discrete(pugi::xml_node node)
+{
+  auto params = get_node_array<double>(node, "parameters");
+
+  std::size_t n = params.size() / 2;
+
+  x_.assign(params.begin(), params.begin() + n);
+  di_ = new DiscreteIndex(params.begin() + n, n)
```

Comment:
If you change `Discrete` to hold a plain `DiscreteIndex` value type, then you could just initialize the `di_` member from the constructor:
```C++
Discrete::Discrete(pugi::xml_node node) : di_{node}
```